nilearn/plotting/displays.py

```diff
@@ -526,11 +526,8 @@ def _map_show(self, img, type='imshow',
 
         if threshold is not None:
             data = img.get_data()
-            if threshold == 0:
-                data = np.ma.masked_equal(data, 0, copy=False)
-            else:
-                data = np.ma.masked_inside(data, -threshold, threshold,
-                                           copy=False)
+            mask = np.abs(data) <= threshold
```

Comment:
Funny you should say that, since there is indeed a bug I bumped into while trying to tweak the test I added. The colorbar code is where it blows up but I believe the problem is more `vmin = -vmax` in plot_stat_map. There is also a `-offset` somewhere in the colorbar code to be fair.

Maybe converting to float before plotting is the simpler option, after all ?